tests/ext-activitylog.test.js

```diff
@@ -27,28 +27,715 @@ function observeChange(element) {
   });
 }
 
-// This applies for other filter options, such as - view type,
-// api type, api name
-test('show/hide logs associated with extension id that is checked/unchecked from filter option', async () => {
-  expect(window.customElements.get('log-view')).toBe(LogView);
-  expect(window.customElements.get('filter-option')).toBe(FilterOption);
-  expect(window.customElements.get('filter-timestamp')).toBe(FilterTimestamp);
+afterEach(() => {
+  history.replaceState(null, null, document.location.origin);
+});
+
+describe('filtering logs from the logs table', () => {
+  describe('filtering logs with checking/unchecking checkboxes', () => {
+    // This applies to filter options as well, such as - extension id, view type, api type, api name
+    test('show/hide logs associated with extension id that is checked/unchecked from filter option', async () => {
+      expect(window.customElements.get('log-view')).toBe(LogView);
+      expect(window.customElements.get('filter-option')).toBe(FilterOption);
+      expect(window.customElements.get('filter-timestamp')).toBe(
+        FilterTimestamp
+      );
+
+      const logs = [
+        {
+          /* renders the 1st row in the table */
+          id: 'id1@test',
+          viewType: 'viewType1@test',
+          type: 'type1@test',
+          data: { test: 'test1@data' },
+          timeStamp: 1597686226302,
+        },
+        {
+          /* renders the 2nd row in the table */
+          id: 'id2@test',
+          viewType: 'viewType2@test',
+          type: 'type2@test',
+          data: { test: 'test2@data' },
+          timeStamp: 1597686226302,
+        },
+      ];
+
+      const addListener = jest.fn();
+      const sendMessage = jest.fn();
+      const connect = jest.fn();
+
+      window.browser = {
+        runtime: {
+          onMessage: { addListener },
+          sendMessage,
+          connect,
+        },
+      };
+
+      sendMessage.mockImplementation(() => {
+        return Promise.resolve({ existingLogs: [] });
+      });
```

Comment:
@rpl I want to have a common the `browser` object that serves all the test cases and and may be set it with `beforeAll` to run for a  single time before running all the test cases.